finality-aleph/src/party/aggregator.rs

```diff
@@ -23,17 +24,71 @@ pub struct IO<B: Block> {
     pub justifications_for_chain: mpsc::UnboundedSender<JustificationNotification<B>>,
 }
 
-async fn run_aggregator<B, C, N>(
-    mut aggregator: BlockSignatureAggregator<'_, B, N, KeyBox>,
+type Rmc<'a, B> = RMCWrapper<'a, <B as Block>::Hash, KeyBox>;
+
+async fn process_new_block_data<'a, B, N>(
```

Comment:
The addition of two functions here is somewhat worrying as the `party::foo` files for tasks were supposed to be very minimal things with close to no logic. But I guess you only dispelled the illusion in the case of aggregator. :/

Anyway, it would be nice to move some of this logic out of here, but I suspect it's out of scope for this PR – what do you think?

Reply:
As per our discussion: aggregator seems to care a bit too much about the networking layer, while it could be a simple piece of _almost_ pure logic. Even though this PR gets simpler after addressing the comments above, I still think this refactor deserves a subsequent PR.

Reply:
I agree with both of you - most of the code from here that is not responsible for creation and launching should be moved to `aggregator` module, but I admit that it could be done in another PR

Reply:
https://cardinal-cryptography.atlassian.net/browse/AZ-645